maps/RandomZLevels/wildwest_props.dm

```diff
@@ -0,0 +1,324 @@
+		/obj/item/ammo_storage/box/a357 = 20
+		)
+
+/obj/item/weapon/card/id/deputy
+	name = "deputy badge"
+	desc = "A metal star that signifies one as a friend of Old Zounds. You're my favorite deputy."
+	assignment = "Deputy"
+	icon_state = "deputystar"
+	//access = list(access_deputy)
+	show_biometrics = FALSE
+
+/obj/machinery/media/jukebox/folk
+	name = "Old Timey Jukebox"
+
+	change_cost = 0
+
+	playlist_id="folk"
+	// Must be defined on your server.
+	playlists=list(
+		"folk" = "House Specials"
+	)
+
+/obj/structure/uraninitecrystal
+	name = "glowing crystal"
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "crystal"
+	light_color = "#00FF00"
+	anchored = TRUE
+	var/lit = FALSE
+
+/obj/structure/uraninitecrystal/New()
+	..()
+	set_light(2, l_color = light_color)
+
+/obj/structure/uraninitecrystal/bullet_act()
+	rad_pulse()
+
+/obj/structure/uraninitecrystal/kick_act()
+	rad_pulse()
+
+/obj/structure/uraninitecrystal/ex_act()
+	rad_pulse()
+
+/obj/structure/uraninitecrystal/proc/rad_pulse()
+	if(lit)
+		return
+	lit = TRUE
+	set_light(6, l_color = light_color)
+	sleep(10 SECONDS)
+	set_light(2, l_color = light_color)
+	lit = FALSE
+	emitted_harvestable_radiation(get_turf(src), 20, range = 5)
+	for(var/mob/living/carbon/M in view(src,3))
+		var/rads = 50 * sqrt( 1 / (get_dist(M, src) + 1) )
+		M.apply_radiation(round(rads/2),RAD_EXTERNAL)
+
+/obj/item/weapon/reagent_containers/food/drinks/drinkingglass/dorf/New()
+	..()
+	reagents.add_reagent(MANLYDORF, 50)
+	on_reagent_change()
+
+/obj/structure/cartrail
+	name = "rail"
+	desc = "A hunk of shaped metal."
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "rail"
+
+/obj/structure/rustycart
+	name = "rusty cart"
+	desc = "This isn't going anywhere fast."
+	//icon = 'icons/obj/vehicles.dmi'
+	//icon_state = "mining_cart"
+	icon = 'icons/obj/storage/storage.dmi'
+	icon_state = "miningcar"
+	anchored = TRUE
+	density = TRUE
+
+/obj/structure/flora/desert
+	icon = 'icons/obj/flora/ausflora.dmi'
+	shovelaway = TRUE
+
+/obj/structure/flora/desert/barrelcactus
+	name = "barrel cactus"
+	desc = "That's a barrel. Wait, no."
+	anchored = TRUE
+	icon_state = "barrelcactus_1"
+
+/obj/structure/flora/desert/barrelcactus/New()
+	..()
+	icon_state = "barrelcactus_[rand(1,2)]"
+
+/obj/structure/flora/desert/barrelcactus/Crossed(atom/movable/AM)
+	..()
+	if(iscarbon(AM))
+		var/mob/living/carbon/L = AM
+		L.reagents.add_reagent(FEVERFEW,3) //This will take 15 ticks to clear, doing about 22 brute (but brute regens easily)
+		to_chat(L, "<span class='danger'>You prick yourself on \the [src].</span>")
+
+/obj/structure/flora/desert/saguaro
+	name = "saguaro cactus"
+	desc = "The space saguaro gets its name from the Earth saguaro, which comes from an indigenous Opata word that refers to saguaros."
+	density = TRUE
+	pass_flags_self = PASSTABLE | PASSGLASS
+	anchored = TRUE
+	icon_state = "saguaro_1"
+
+/obj/structure/flora/desert/saguaro/Bumped(atom/movable/AM)
```

Comment:
how come you use crossed on the barrelcactus but bumped on this one?
is it because of the density?

Reply:
yes, the saguaro blocks movement and the barrel cactus does not